server/routes/ops.js

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var Q = require('q');
+var pg = require('pg');
+var es = require('elasticsearch');
+
+var config = require('../config');
+
+var dbParams = {
+  user: config.get('db.postgres.user'),
+  password: config.get('db.postgres.password'),
+  host: config.get('db.postgres.host'),
+  port: config.get('db.postgres.port'),
+  database: config.get('db.postgres.database'),
+  ssl: config.get('db.postgres.ssl')
+};
+
+// >:-(
+// https://github.com/elasticsearch/elasticsearch-js/issues/33
+var esParamsFactory = function() {
+  return {
+    host: {
+      host: config.get('db.elasticsearch.host'),
+      post: config.get('db.elasticsearch.port'),
+      log: 'verbose'
+    }
+  };
+};
+
+var pgCheck = function () {
+  var deferred = Q.defer();
+  pg.connect(dbParams, function(err, client, done) {
+    done(); // Close database connection
+    if (err) {
+      return deferred.reject(err);
+    }
+    deferred.resolve();
+  });
+  return deferred.promise;
+};
+
+var esCheck = function () {
+  var deferred = Q.defer();
+  var esClient = new es.Client(esParamsFactory());
+  esClient.search({
+    index: 'chronicle',
+    type: 'visits',
+    size: 0,
+    body: {}
+  }).then(deferred.resolve, deferred.reject).catch(deferred.reject);
+  return deferred.promise;
+};
+
+module.exports = [{
+  method: 'GET',
+  path: '/__heartbeat__',
+  config: {
+    handler: function (request, reply) {
+      Q.all([
+        pgCheck(),
+        esCheck()
+      ]).then(function (results) {
+        reply('ok').code(200);
+      }, function (err) {
+        reply(err.message).code(503);
+      }).catch(function (err) {
+        reply(err.message).code(503);
```

Comment:
I'm not sure I understand what the trailing `catch` is doing on line 70. It looks like the idea is to catch a possible error thrown by the `onFulfilled` handler on line 67. But the catch handler is calling the same code as that function, so if it failed on line 67, it'll fail again on line 71, so we don't need it (I think).

It might be better to terminate the chain using `done` in place of `then` on line 66, and skip the trailing `catch`.